openqaoa/workflows/optimizer.py

```diff
@@ -430,42 +536,206 @@ def compile(self, problem: QUBO = None, verbose: bool = True):
-                device=self.qaoa.device,
-                params_type=self.qaoa.circuit_properties.param_type,
-                init_type=self.qaoa.circuit_properties.init_type,
-                optimizer_dict=self.qaoa.classical_optimizer.asdict(),
-                backend_properties=self.qaoa.backend_properties.__dict__)
-        elif self.rqaoa_parameters.rqaoa_type == 'custom':
-            res = custom_rqaoa(
-                hamiltonian=self.qaoa.cost_hamil,
-                mixer=self.rqaoa_mixer,
-                p=self.qaoa.circuit_properties.p,
-                n_cutoff=self.rqaoa_parameters.n_cutoff,
-                steps=self.rqaoa_parameters.steps,
-                device=self.qaoa.device,
-                params_type=self.qaoa.circuit_properties.param_type,
-                init_type=self.qaoa.circuit_properties.init_type,
-                optimizer_dict=self.qaoa.classical_optimizer.asdict(),
-                backend_properties=self.qaoa.backend_properties.__dict__)
+
+        # lists to append the eliminations and the qaoa objects
+        elimination_tracker = []
+        qaoa_steps = []
+        problem_steps = []
+
+        # get variables
+        problem = self.problem  
+        n_cutoff = self.rqaoa_parameters.n_cutoff
+        n_qubits = problem.n
+        counter = self.rqaoa_parameters.counter 
+
+        # create a different max_terms function for each type 
+        if self.rqaoa_parameters.rqaoa_type.lower() == "adaptive":
+            f_max_terms = rqaoa.ada_max_terms  
         else:
-            raise f'rqaoa_type {self.rqaoa_parameters.rqaoa_type} is not supported. Please selet either "adaptive" or "custom'
+            f_max_terms = rqaoa.max_terms #? rqaoa.max_terms -> rqaoa.custom_max_terms ?
 
-        self.result = res
-        return None
+        # If above cutoff, loop quantumly, else classically
+        while n_qubits > n_cutoff:
 
-    def asdict(self):
-        attributes_dict = convert2serialize(self)
-        return attributes_dict
+            # Run QAOA
+            self._q.optimize()
+
+            # Obtain statistical results
+            exp_vals_z, corr_matrix = self._exp_val_hamiltonian_termwise()
+            # Retrieve highest expectation values according to adaptive method
+            max_terms_and_stats = f_max_terms(exp_vals_z, corr_matrix, self._n_step(n_qubits, n_cutoff, counter))
+            # Generate spin map
+            spin_map = rqaoa.spin_mapping(problem, max_terms_and_stats)
+            # Eliminate spins and redefine problem
+            new_problem, spin_map = rqaoa.redefine_problem(problem, spin_map)
+
+            # Extract final set of eliminations with correct dependencies and update tracker
+            eliminations = {(spin_map[spin][1],spin):spin_map[spin][0] for spin in sorted(spin_map.keys()) if spin != spin_map[spin][1]}
+            elimination_tracker.append(eliminations)
+
+            # Extract new number of qubits
+            n_qubits = new_problem.n
```

Comment:
Here you overwrite the variable defined in line `593`. I'd use a different variable name instead

Reply:
The `n_qubits` variable? but I need to change it to update the number of qubits with the new problem